src/wal/reader.rs

```diff
@@ -213,6 +215,12 @@ where
             window: iter,
         })
     }
+
+    /// Extract lookahead window from Genesis.
+    pub fn lookahead_from_genesis(genesis: &Genesis) -> u64 {
+        ((3.0 * genesis.byron.protocol_consts.k as f32)
+            / (genesis.shelley.active_slots_coeff.unwrap())) as u64
+    }
```

Comment:
doesn't belong here. The ledger should provide a function describing the "mutability window" from genesis data.

Reply:
I think there's already a method in there.